src/thread.rs

```diff
@@ -415,3 +411,5 @@ impl<'env> Drop for Scope<'env> {
         self.drop_all().unwrap();
     }
 }
+
+type ScopedThreadResult<T> = Arc<Mutex<Option<T>>>;
```

Comment:
After inserting a newline here, I think we're ready to merge this PR.  (Perhaps we can deal with the newline issue in another PR, too.)  @oliver-giersch @stjepang are you okay with merging?